packages/amplify-graphql-api-construct-tests/src/__tests__/migration/base-migration.test.ts

```diff
@@ -0,0 +1,157 @@
+import * as path from 'path';
+import {
+  createNewProjectDir,
+  deleteProjectDir,
+  deleteProject,
+} from 'amplify-category-api-e2e-core';
+import { initCDKProject, cdkDeploy, cdkDestroy, createGen1ProjectForMigration, writeTableMap, deleteDDBTables } from '../../commands';
+import { graphql } from '../../graphql-request';
+import { DURATION_20_MINUTES } from '../../utils/duration-constants';
+
+jest.setTimeout(DURATION_20_MINUTES);
+
+describe('Base Migration', () => {
+  let gen1ProjRoot: string;
+  let gen2ProjRoot: string;
+  let gen1ProjFolderName: string;
+  let gen2ProjFolderName: string;
+  let dataSourceMapping: Record<string, string>;
+
+  beforeEach(async () => {
+    gen1ProjFolderName = 'basemigrationgen1';
+    gen2ProjFolderName = 'basemigrationgen2';
+    gen1ProjRoot = await createNewProjectDir(gen1ProjFolderName);
+    gen2ProjRoot = await createNewProjectDir(gen2ProjFolderName);
+  });
+
+  afterEach(async () => {
+    try {
+      await deleteProject(gen1ProjRoot);
+    } catch (_) {
+      /* No-op */
+    }
+    try {
+      await cdkDestroy(gen2ProjRoot, '--all');
+    } catch (_) {
+      /* No-op */
+    }
+
+    try {
+      // Tables are set to retain when migrating from gen 1 to gen 2
+      await deleteDDBTables(Object.values(dataSourceMapping));
+    } catch (_) {
+      /* No-op */
+    }
+
+    deleteProjectDir(gen1ProjRoot);
+    deleteProjectDir(gen2ProjRoot);
+  });
+
+  test('base migration', async () => {
+    const {
+      GraphQLAPIEndpointOutput: gen1APIEndpoint,
+      GraphQLAPIKeyOutput: gen1APIKey,
+      DataSourceMappingOutput,
+    } = await createGen1ProjectForMigration(gen1ProjFolderName, gen1ProjRoot, 'simple_model_public_auth.graphql');
+    dataSourceMapping = JSON.parse(DataSourceMappingOutput);
+    const templatePath = path.resolve(path.join(__dirname, '..', 'backends', 'migration', 'base'));
+    const name = await initCDKProject(gen2ProjRoot, templatePath);
+    writeTableMap(gen2ProjRoot, DataSourceMappingOutput);
+    const outputs = await cdkDeploy(gen2ProjRoot, '--all');
+    const { awsAppsyncApiEndpoint: gen2APIEndpoint, awsAppsyncApiKey: gen2APIKey } = outputs[name];
+
+    const gen1Result = await graphql(
+      gen1APIEndpoint,
+      gen1APIKey,
+      /* GraphQL */ `
+        mutation CREATE_TODO {
+          createTodo(input: { content: "todo desc" }) {
+            id
+            content
+          }
+        }
+      `,
+    );
+    expect(gen1Result.statusCode).toEqual(200);
```

Comment:
Since AppSync returns a 200 even if the result has GraphQL errors, let's add a comment that we'll assert the actual results of the create by inspecting the `list` results below. Presumably this assertion is just to make sure we don't get a 400 b/c of a malformed request?